simln-lib/src/sim_node.rs

```diff
@@ -992,7 +1004,7 @@ async fn remove_htlcs(
     route: Path,
     payment_hash: PaymentHash,
     success: bool,
-) -> Result<(), ForwardingError> {
+) -> ForwardResult {
```

Comment:
I think that this can just be `Result<(), CriticalError>` - we only hit critical failures here.